From 1a2ad64be919c444d462514ef87e9c13a0b734ed Mon Sep 17 00:00:00 2001
From: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Date: Wed, 3 Oct 2018 11:58:54 +0300
Subject: [PATCH 4768/5242] MLK-19797 sound: asoc: fix clk_div computed value

commit  456406762cd0290e7471ee94e61b82751660ead0 from
https://source.codeaurora.org/external/imx/linux-imx.git

When clk_div is computed, the K factor from documentation is ignored.
It happened to work on medium quality because the K factor was 1.
Now the clk_div is computed accordingly to the documentation.

Signed-off-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Reviewed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_micfil.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/sound/soc/fsl/fsl_micfil.c b/sound/soc/fsl/fsl_micfil.c
index 672b628..94c12d9 100644
--- a/sound/soc/fsl/fsl_micfil.c
+++ b/sound/soc/fsl/fsl_micfil.c
@@ -749,18 +749,18 @@ static inline int get_pdm_clk(struct fsl_micfil *micfil,
 
 	switch (qsel) {
 	case MICFIL_HIGH_QUALITY:
-		bclk = rate * 8 * osr;
+		bclk = rate * 8 * osr / 2; /* kfactor = 0.5 */
 		break;
 	case MICFIL_MEDIUM_QUALITY:
 	case MICFIL_VLOW0_QUALITY:
-		bclk = rate * 4 * osr;
+		bclk = rate * 4 * osr * 1; /* kfactor = 1 */
 		break;
 	case MICFIL_LOW_QUALITY:
 	case MICFIL_VLOW1_QUALITY:
-		bclk = rate * 2 * osr;
+		bclk = rate * 2 * osr * 2; /* kfactor = 2 */
 		break;
 	case MICFIL_VLOW2_QUALITY:
-		bclk = rate * osr;
+		bclk = rate * osr * 4; /* kfactor = 4 */
 		break;
 	default:
 		dev_err(&micfil->pdev->dev,
-- 
1.7.9.5

